### DISCUSSION TITLE
no events raised after setting up connection (question / help request)

### DISCUSSION THREAD

**Discussion by samthomson** [enisdenjo/graphql-ws#212] [Q&A] [answered]
I can create a connection and send queries over it, however I don't see/get any events firing as I'd expect (which I'd like, to feed into app state regarding connected status). 
I am creating the client connection in a redux middleware:
```
let socket: any = null

const socketMiddleware = () => 
  {
	return (next: any) => async (action: WSAction) => {
		switch (action.type) {
			case WSActionType.connect:
				console.log('WSActionType.connect')
				const url = `ws://${window.location.hostname}:3501/wsgraphql`

				socket = createClient({
					url,
					on: {
						closed: () => {
						  console.log("closed");
						},
						connected: () => {
						  console.log("connected");
						},
						opened: () => {
						  console.log("opened ");
						},
						error: (e) => {
						  console.log("error ", e);
						},
					  }
				})				
				break
```
In the above code I get the `WSActionType.connect` message in log, but none of those defined in `on` of `createClient`.
Do you see something wrong? Or is there a better approach to creating the connection in redux middleware (this is how I've done it before for non-graphql data)?

Thanks for the great library.

**Comment by enisdenjo** [comment, marked as answer]
The library is in [`lazy` mode](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/client.clientoptions.md#lazy) by default; meaning, it connects on demand (connect on first subscribe and disconnect on last unsubscribe). For you to get those logs as is, you have to subscribe or execute a GQL operation with the client.

If you want the client to connect as soon as you create it, use the `lazy: false` option.
